docs/source/reference/modding/extended.rst

```diff
@@ -0,0 +1,212 @@
+These features can conflict with mods, and such situations should be handled as any other mod conflict by patching the mods themselves.
+The engine itself does not provide any kind of blacklisting of incompatible mods.
+
+
+Native herbalism support
+------------------------
+
+In OpenMW it is possible to add one or more NiSwitchNodes with a ``HerbalismSwitch`` name.
+Every switch node should have at least two child nodes (the first one represents the unharvested container, the second one - the harvested container).
+If an organic container's mesh has such nodes, it is considered to be a plant. During activation, a window with the plant's content is not shown,
+OpenMW transfers the contents directly into the player's inventory, triggers a theft crime event if the plant is owned and toggles the harvested model state.
+
+It is also possible to use scripts with ``OnActivate`` command for such containers. For example, when player needs a tool to harvest a plant (e.g. a pickaxe for ore).
+
+Keep in mind that the collision detection system ignores switch nodes, so add a ``RootCollisionNode`` or ``NCO`` NiStringExtraData to harvestable meshes.
+
+Advantages of described approach over old herbalism mods:
+
+1. There is no need to spawn separate "harvested" objects
+
+2. There is no need to attach a script to every container
+
+3. It supports an ownership check (the original engine without MWSE does not)
+
+4. It does not alter original respawn mechanics
+
+An example of mod which uses this feature is `Graphic Herbalism`_.
+
+Animated containers support
+---------------------------
+
+It is possible to attach opening/closing animations for containers. To do this, you need to create a KF-file for the container with the following groups:
+
+1. ``Idle`` (with ``Start`` and ``Stop`` keys)
+
+2. ``ContainerOpen`` (with ``Start``, ``Loot`` and ``Stop`` keys)
+
+3. ``ContainerClose`` (with ``Start`` and ``Stop`` keys)
+
+The ``Idle`` group can have the zero time, the ``Loot`` key for ``ContainerOpen`` allows to play a part of opening animation in the background.
+
+For example, with the following setup, the opening animation has 1.0 sec duration and shows the container window in the middle of opening animation:
+
+::
+
+    0.0: ContainerOpen: start
+    0.5: ContainerOpen: loot
+    1.0: ContainerOpen: stop
+
+It is also possible to attach opening/closing sounds to container's animations:
+
+::
+
+    1.0: ContainerClose: open
+    1.01: Sound: AC_dw_drawer_close
+    2.0: ContainerClose: stop
+
+The text key approach is the same as the one used for sound playback in animations in general
+Note that the sound starting time value is slightly higher than the closing animation start, otherwise sound will be played at the end of opening animation in this example.
+
+It is important to assign a RootCollisionNode to the container mesh -- the collision shape will depend on the animation state otherwise, and this can have a performance impact.
```

Comment:
```suggestion
It is important to assign a RootCollisionNode to the container mesh; otherwise the collision shape will depend on the animation state, and this can have a performance impact.
```